File: Medium/2966. Divide Array Into Arrays With Max Difference.py
from typing import List


class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        nums.sort()
        n = len(nums)
        for i in range(0, n, 3):
            if nums[i + 2] - nums[i] > k:
                return []
        return [[nums[i], nums[i + 1], nums[i + 2]] for i in range(0, n, 3)]

File: Medium/2966. Divide Array Into Arrays With Max Difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        Arrays.sort(nums);
        int[][] divided = new int[nums.length / 3][3];
        for (int r = 0; r < divided.length; r++) {
            if (nums[r * 3 + 2] - nums[r * 3] > k) {
                return new int[][] {};
            }
            for (int c = 0; c < 3; c++) {
                divided[r][c] = nums[r * 3 + c];
            }
        }
        return divided;
    }
}